File: fortran_iv/THETAG.f
      FUNCTION THETAG(EP)
      COMMON /E1/XMO,XNODEO,OMEGAO,EO,XINCL,XNO,XNDT2O,XNDD6O,BSTAR,
     1 X,Y,Z,XDOT,YDOT,ZDOT,EPOCH,DS50
      DOUBLE PRECISION EPOCH,D,THETA,TWOPI,YR,TEMP,EP,DS50
      TWOPI=6.28318530717959D0
      YR=(EP+2.D-7)*1.D-3
      JY=YR
      YR=JY
      D=EP-YR*1.D3
      IF(JY.LT.10) JY=JY+80
      N=(JY-69)/4
      IF(JY.LT.70) N=(JY-72)/4
      DS50=7305.D0 + 365.D0*(JY-70) +N + D
      THETA=1.72944494D0 + 6.3003880987D0*DS50
      TEMP=THETA/TWOPI
      I=TEMP
      TEMP=I
      THETAG=THETA-TEMP*TWOPI
      IF(THETAG.LT.0.D0) THETAG=THETAG+TWOPI
      RETURN
      END

File: c/thetag.c
#include "str3.h"

double thetag(struct E1 * const E1, const double ep) {
  const double twopi = 6.28318530717959;
  double yr = (ep + 2E-7) * 1E-3;
  int jy = (int) yr;
  yr = jy;
  const double d = ep - yr * 1E3;
  if (jy < 10) jy = jy + 80;
  int n = (jy - 69) / 4;
  if (jy < 70) n = (jy - 72) / 4;
  E1->ds50 = 7305. + 365. * (jy - 70) + n + d;
  const double theta = 1.72944494 + 6.3003880987 * E1->ds50;
  double temp = theta / twopi;
  const int i = (int) temp;
  temp = i;
  double thetag = theta - temp * twopi;
  if (thetag < 0.) thetag = thetag + twopi;
  return thetag;
}

